dev/RadialGradientBrush/RadialGradientBrush.cpp

```diff
@@ -0,0 +1,228 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "common.h"
+#include "Vector.h"
+#include "SharedHelpers.h"
+#include "RadialGradientBrush.h"
+#include "RuntimeProfiler.h"
+#include "ResourceAccessor.h"
+
+RadialGradientBrush::RadialGradientBrush()
+{
+    __RP_Marker_ClassById(RuntimeProfiler::ProfId_RadialGradientBrush);
+
+    m_gradientStops = winrt::make<Vector<winrt::GradientStop>>().as<winrt::IObservableVector<winrt::GradientStop>>();
+}
+
+winrt::IObservableVector<winrt::GradientStop> RadialGradientBrush::GradientStops()
+{
+    return m_gradientStops;
+}
+
+void RadialGradientBrush::OnConnected()
+{
+    // XCBB will use fallback rendering in design v1 mode, so do not create a CompositionBrush.
+    if (SharedHelpers::IsInDesignMode()) { return; }
+
+    m_isConnected = true;
+
+    EnsureCompositionBrush();
+
+    if (SharedHelpers::IsCompositionRadialGradientBrushAvailable())
+    {
+        // If CompositionRadialGradientBrush will be used then listen for changes to gradient stops so the composition brush can be updated.
+        m_gradientStopsChangedToken = m_gradientStops.VectorChanged({
+            [this](winrt::IObservableVector<winrt::GradientStop> const& sender, winrt::IVectorChangedEventArgs const& args)
+            {
+                if (m_brush)
+                {
+                    UpdateCompositionGradientStops();
+                }
+            }
+            });
+    }
+    else
+    {
+        // If CompositionRadialGradientBrush won't be used then listen for changes to the fallback color so the fallback brush can be updated.
+        m_fallbackColorChangedToken.value = RegisterPropertyChangedCallback(winrt::XamlCompositionBrushBase::FallbackColorProperty(), { this, &RadialGradientBrush::OnFallbackColorChanged });
+    }
+}
+
+void RadialGradientBrush::OnDisconnected()
+{
+    m_isConnected = false;
+
+    if (m_brush)
+    {
+        m_brush.Close();
+        m_brush = nullptr;
+        CompositionBrush(nullptr);
+    }
+
+    if (m_gradientStopsChangedToken)
+    {
+        m_gradientStops.VectorChanged(m_gradientStopsChangedToken);
+    }
+
+    if (m_fallbackColorChangedToken)
+    {
+        UnregisterPropertyChangedCallback(winrt::XamlCompositionBrushBase::FallbackColorProperty(), m_fallbackColorChangedToken.value);
+        m_fallbackColorChangedToken.value = 0;
+    }
+}
+
+void RadialGradientBrush::OnPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    if (SharedHelpers::IsCompositionRadialGradientBrushAvailable())
+    {
+        auto property = args.Property();
```

Comment:
>auto [](start = 8, length = 4)

c++ guidelines Con.4 ([https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rconst-const](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rconst-const)) says this should be const